diff --git a/SConstruct b/SConstruct
index 5965b53..50a76ef 100644
--- a/SConstruct
+++ b/SConstruct
@@ -320,7 +320,7 @@ add_option('propagate-shell-environment',
            0, False)
 
 if darwin:
-    osx_version_choices = ['10.6', '10.7', '10.8', '10.9']
+    osx_version_choices = ['10.6', '10.7', '10.8', '10.9', '10.10']
     add_option("osx-version-min", "minimum OS X version to support", 1, True,
                type = 'choice', default = osx_version_choices[0], choices = osx_version_choices)
 
