diff --git a/configure b/configure
index ae8fd6b..55b2e4e 100755
--- a/configure
+++ b/configure
@@ -16720,7 +16720,7 @@ ac_res=$ac_cv_search_SSL_library_init
 if test "$ac_res" != no; then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
 
-      LIBS="$LIBS -lssl"
+      LIBS="$LIBS -lssl -lcrypto"
 
      { $as_echo "$as_me:${as_lineno-$LINENO}: checking Openssl library version >= 1.0.0c" >&5
 $as_echo_n "checking Openssl library version >= 1.0.0c... " >&6; }
diff --git a/imap/Makefile b/imap/Makefile
index ec1ded1..d9d2bd1 100644
--- a/imap/Makefile
+++ b/imap/Makefile
@@ -414,25 +414,25 @@ lsu:	an
 osi:	an
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=osx IP=$(IP6) CC=arm-apple-darwin-gcc \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private"
+	SPECIALS="SSLINCLUDE=__PREFIX__/include/openssl SSLLIB=__PREFIX__/lib GSSINCLUDE=__PREFIX__/include GSSLIB=__PREFIX__/lib PAMDLFLAGS=-lpam"
 
 oxp:	an
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=osx IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
 	PASSWDTYPE=pam \
 	EXTRACFLAGS="$(EXTRACFLAGS) -DMAC_OSX_KLUDGE=1" \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib PAMDLFLAGS=-lpam"
+	SPECIALS="SSLINCLUDE=__PREFIX__/include/openssl SSLLIB=__PREFIX__/lib GSSINCLUDE=__PREFIX__/include GSSLIB=__PREFIX__/lib PAMDLFLAGS=-lpam"
 
 oxs:	an
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=osx IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
 	PASSWDTYPE=pam \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib PAMDLFLAGS=-lpam"
+	SPECIALS="SSLINCLUDE=__PREFIX__/include/openssl SSLLIB=__PREFIX__/lib GSSINCLUDE=__PREFIX__/include GSSLIB=__PREFIX__/lib PAMDLFLAGS=-lpam"
 
 osx:	osxok an
 	$(TOUCH) ip6
 	$(BUILD) BUILDTYPE=$@ IP=$(IP6) EXTRAAUTHENTICATORS="$(EXTRAAUTHENTICATORS) gss" \
-	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib"
+	SPECIALS="SSLINCLUDE=__PREFIX__/include/openssl SSLLIB=__PREFIX__/lib GSSINCLUDE=__PREFIX__/include GSSLIB=__PREFIX__/lib PAMDLFLAGS=-lpam"
 
 osxok:
 	@$(SH) -c '(test ! -f /usr/include/pam/pam_appl.h ) || make osxwarn'

