From 86d684b4228420064c1e0e925732e208ad548e3c Mon Sep 17 00:00:00 2001
From: Dominyk Tiller <dominyktiller@gmail.com>
Date: Mon, 3 Nov 2014 23:09:43 +0000
Subject: yosemiteyo

---
 alpine/alpine.c | 2 +-
 pith/util.h     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/alpine/alpine.c b/alpine/alpine.c
index 9e7298f..4440404 100644
--- a/alpine/alpine.c
+++ b/alpine/alpine.c
@@ -3322,7 +3322,7 @@ pine_gets_bytes(int reset)
   This is also called from imap routines and fs_get and fs_resize.
   ----*/
 void
-panic(char *message)
+panic(const char *message)
 {
     char buf[256];
 
diff --git a/pith/util.h b/pith/util.h
index defbca6..de96283 100644
--- a/pith/util.h
+++ b/pith/util.h
@@ -56,7 +56,7 @@ int           *cpyint(int);
 /* currently mandatory to implement stubs */
 
 /* called when we detect a serious program error */
-void	  panic(char *);
+void	  panic(const char *, ...);
 
 /* called when testing to see if panic state is in effect */
 int	 panicking(void);
-- 
2.1.3

