diff --git a/SConstruct b/SConstruct
index dc6698e..bc0b5f6 100644
--- a/SConstruct
+++ b/SConstruct
@@ -35,6 +35,9 @@ AddOption('--prefix', dest='prefix', type='string', nargs=1, action='store', met
 prefix = GetOption('prefix')
 env = Environment(options = opts)
 
+env.Append(CCFLAGS = ["-Isrc"])
+
+
 if env['RELEASE']:
   env.Append(CPPDEFINES=["PUBLIC_RELEASE"])
   env['DEBUG'] = 0
@@ -138,14 +141,15 @@ else:
       # Should work on any *nix
       env.Append(CCFLAGS = ["-DLUA_USE_LINUX"])
     lua_available = False
-    if conf.CheckLib('lua5.1'):
-      env.Append(LINKFLAGS = ["-ldl", "-llua5.1"])
-      env.Append(CCFLAGS = ["-I/usr/include/lua5.1"])
-      lua_available = True
-    elif conf.CheckLib('lua'):
-      env.Append(LINKFLAGS = ["-ldl", "-llua"])
-      env.Append(CCFLAGS = ["-I/usr/include/lua"])
+    if conf.CheckLib('lua5.2'):
+      env.Append(LINKFLAGS = ["-ldl", "-llua5.2"])
+      env.Append(CCFLAGS = ["-I/usr/include/lua5.2"])
+      env.Append(CPPDEFINES = ["LUA_COMPAT_ALL"])
       lua_available = True
+    elif conf.CheckLib('lua5.1'):
+       env.Append(LINKFLAGS = ["-ldl", "-llua5.1"])
+       env.Append(CCFLAGS = ["-I/usr/include/lua5.1"])
+       lua_available = True
     if lua_available == False:
       print 'Could not find liblua, exiting!'
       Exit(1)
diff --git a/src/SConscript b/src/SConscript
index 4713e15..deff2c8 100644
--- a/src/SConscript
+++ b/src/SConscript
@@ -1,6 +1,7 @@
 import glob
 file_list = glob.glob('*.cpp')
-file_list.remove('lua-engine.cpp') # use logic below for this
+file_list.remove('lua-engine.cpp') # use lua logic below for this
+file_list.remove('lua-compatibility.cpp')
 
 subdirs = Split("""
 boards
diff --git a/src/drivers/sdl/input.cpp b/src/drivers/sdl/input.cpp
index c371b2f..045c394 100644
--- a/src/drivers/sdl/input.cpp
+++ b/src/drivers/sdl/input.cpp
@@ -426,6 +426,10 @@ void FCEUD_LoadStateFrom ()
 	FCEUI_LoadState (fname.c_str ());
 }
 
+#define KEYBOARDTRANSFORMER_SPECIFIC
+#ifdef KEYBOARDTRANSFORMER_SPECIFIC
+/**
+
 /**
 * Hook for transformer board
 */
diff --git a/src/lua-engine.cpp b/src/lua-engine.cpp
index 53774f7..fc21044 100644
--- a/src/lua-engine.cpp
+++ b/src/lua-engine.cpp
@@ -24,6 +24,7 @@
 #include "utils/xstring.h"
 #include "utils/memory.h"
 #include "fceulua.h"
+#include "lua-compatibility.h"
 
 #ifdef WIN32
 #include "drivers/win/common.h"
@@ -273,6 +274,7 @@ static void FCEU_LuaOnStop()
 	gui_used = GUI_CLEAR;
 	//if (wasPaused && !FCEUI_EmulationPaused())
 	//	FCEUI_ToggleEmulationPause();
+	//if (fps_scale != 256)							//thanks, we already know it's on normal speed
 	FCEUD_SetEmulationSpeed(EMUSPEED_NORMAL);		//TODO: Ideally lua returns the speed to the speed the user set before running the script
 													//rather than returning it to normal, and turbo off.  Perhaps some flags and a FCEUD_GetEmulationSpeed function
 	turbo = false;
